### PR TITLE
Improvements to the plotting guide

### PR DESCRIPTION
**Pull request by dstansby**
I didn't get all the way through the guide, but I found this diff hanging around where I've:
- Changed lots to 1 sentence per line.
- Improved the bits I updated

### CODE REVIEW

File: docs/guide/plotting.rst [hunk 1]
```diff
@@ -88,30 +72,23 @@ and access the figures and axes objects. This is shown in the following example.
...
-which supports both simple and advanced matplotlib usage. The following examples
-focus on the map object but they should be applicable across all of the data
-objects.
+Plotting in sunpy
+*****************
 
-4.1 peek()
-**********
+To be consistent with Matplotlib, sunpy has developed a standard plotting interface which supports both simple and advanced Matplotlib usage.
+The following examples focus on the map object, but both ``peek()`` ``plot()`` work on time series objects too.
```
Comment (nabobalis) [inline docs/guide/plotting.rst:85]:
```suggestion
The following examples focus on the map object, but both ``peek()`` and ``plot()`` work on time series objects too.
```

File: docs/guide/plotting.rst [hunk 2]
```diff
@@ -4,31 +4,24 @@
...
-2. Pyplot
-*********
+sunpy makes use of `Matplotlib <https://matplotlib.org/>`_ for all of its plotting - as such, it tries to follow the Matplotlib plotting philosophy.
+We'll start by going over the basics of plotting in Matplotlib.
+If you are already familiar with Matplotlib, skip ahead to the second section.
+
+Matplotlib Tutorial
+*******************
+
+This tutorial is a summary of one that can be found in the `Matplotlib usage documentation <https://matplotlib.org/stable/users/explain/api_interfaces.html>`_.
```
Comment (nabobalis) [inline docs/guide/plotting.rst:14]:
```suggestion
This tutorial is a summary of one that can be found in the `Matplotlib usage documentation <https://matplotlib.org/stable/users/explain/api_interfaces.html>`__.
```

File: docs/guide/plotting.rst [hunk 3]
```diff
@@ -39,41 +32,32 @@ Here is a simple example of pyplot usage.
...
-is something that the matplotlib devs get complaints about often and so this may change.
-A discussion about this can be found `here
-<https://stackoverflow.com/questions/5524858/matplotlib-show-doesnt-work-twice>`_.
-Don't be confused by another command called `~matplotlib.pyplot.draw`.
-This is only used while in interactive mode.
+The `~matplotlib.pyplot.show` command opens a plot on the screen and blocks execution until the plot window is closed.
+The `~matplotlib.pyplot.show` command only works once - if you call `~matplotlib.pyplot.show` again after the above code is executed nothing happens.
 
-To turn on interactivity for pyplot use the command ::
+To turn on interactive plotting for pyplot use the command ::
```
Comment (nabobalis) [inline docs/guide/plotting.rst:38]: Is this something we want to users?

File: docs/guide/plotting.rst [hunk 4]
```diff
@@ -88,30 +72,23 @@ and access the figures and axes objects. This is shown in the following example.
...
-**********
+To be consistent with Matplotlib, sunpy has developed a standard plotting interface which supports both simple and advanced Matplotlib usage.
+The following examples focus on the map object, but both ``peek()`` ``plot()`` work on time series objects too.
 
-For quick and easy access to a plot all sunpy base objects (i.e. maps, spectra,
-timeseries) define their own `~sunpy.map.mapbase.GenericMap.peek` command which
-will create a plot for you and show it without you having to deal with any
-matplotlib setup. This is so that it is easy to take a quick look at your data.
-For example you can make the following plot.
+peek()
```
Comment (nabobalis) [inline docs/guide/plotting.rst:87]: I wonder if we want to link these to the GenericMap methods?

Comment (dstansby) [reply docs/guide/plotting.rst:87]: 🤷

File: docs/guide/plotting.rst (hunk 3 above)
Comment (nabobalis) [reply docs/guide/plotting.rst:38]: I will drop it, I just don't like telling users to avoid doing show.

**Review by nabobalis — APPROVED** [review]

**Pull request merged**